pkg/datapath/linux/config/config.go

```diff
@@ -143,7 +143,7 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 	encapProto := option.Config.TunnelProtocol
 	if !option.Config.TunnelingEnabled() &&
 		option.Config.EnableNodePort &&
-		option.Config.NodePortMode == option.NodePortModeDSR &&
+		option.Config.NodePortMode != option.NodePortModeSNAT &&
```

Comment:
This is more concise than `== DSR || == Hybrid` but maybe less future proof?

Reply:
Ack, not a big fan of it either - for now, I believe this is in sync with how the remaining code does it. But I can follow up with some `NodePortUsesDSR()` helper.

Reply:
+1. If we are going to make setting more logical, I also vote for not using the name NodePort to refer to everything related to forwarding traffic